### PR TITLE
[Suricata] Add Observer Metadata

### LINKED ISSUE #6984: [Suricata] Enable `observer.*` Fields for Integration
## Describe the Issue

The Suricata integration does currently support removing `host.*` fields if "forwarded" is in the `tags`, but it does not currently populate the `observer.*` fields.

### PR DESCRIPTION
**Pull request by MakoWish**
## Type of change
- Enhancement

## What does this PR do?

The Suricata integration currently supports removing `host.*` fields if "forwarded" is in `tags`, but it does not currently populate the `observer.*` fields. This PR adds that functionality. 

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have incremented the version my package's `manifest.yml` file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [X] Modify default documents config to include "forwarded" tag
- [X] Verify expected results documents include new `observer.*` fields

## Related issues

- Closes #6984

### CODE REVIEW

File: packages/suricata/data_stream/eve/_dev/test/pipeline/test-common-config.yml [hunk 1]
```diff
@@ -4,3 +4,4 @@ fields:
   "@timestamp": "2020-04-28T11:07:58.223Z"
   tags:
     - preserve_original_event
+    - forwarded
```
Comment (efd6) [inline packages/suricata/data_stream/eve/_dev/test/pipeline/test-common-config.yml:7]: Can we have a test case where this is not set as well?

File: packages/suricata/data_stream/eve/agent/stream/log.yml.hbs [hunk 2]
```diff
@@ -13,7 +13,10 @@ tags:
 {{#contains "forwarded" tags}}
 publisher_pipeline.disable_host: true
 {{/contains}}
-{{#if processors}}
 processors:
+{{#contains "forwarded" tags}}
+  - add_observer_metadata: ~
```
Comment (efd6) [inline packages/suricata/data_stream/eve/agent/stream/log.yml.hbs:18]: This has a significant chance of causing a YAML syntax issue.
```suggestion
- add_observer_metadata: ~
```
Given with this we are adding metadata in the forwarded case, is it worth also adding
```
{{else}}
- add_host_metadata: ~
```

File: packages/suricata/data_stream/eve/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -5,6 +5,18 @@ processors:
   - set:
       field: ecs.version
       value: '8.8.0'
+  - set:
+      if: "(ctx?.tags.contains('forwarded'))"
```
Comment (efd6) [inline packages/suricata/data_stream/eve/elasticsearch/ingest_pipeline/default.yml:9]:
```suggestion
      if: ctx.tags != null && ctx.tags.contains('forwarded')
```
Similar below (`ctx` is always present).

File: packages/suricata/data_stream/eve/_dev/test/pipeline/test-common-config.yml (hunk 1 above)
Comment (MakoWish) [reply packages/suricata/data_stream/eve/_dev/test/pipeline/test-common-config.yml:7]: I'll just set it on a couple and remove it from the common config.

File: packages/suricata/data_stream/eve/elasticsearch/ingest_pipeline/default.yml (hunk 3 above)
Comment (efd6) [reply packages/suricata/data_stream/eve/elasticsearch/ingest_pipeline/default.yml]: This needs to happen on the two below as well.

File: packages/suricata/data_stream/eve/_dev/test/pipeline/test-common-config.yml (hunk 1 above)
Comment (MakoWish) [reply packages/suricata/data_stream/eve/_dev/test/pipeline/test-common-config.yml]: I just set it on `alerts` and `small`, removed from common, and regenerated.

File: packages/winlog/data_stream/winlog/elasticsearch/ingest_pipeline/default.yml
Comment (efd6) [inline packages/winlog/data_stream/winlog/elasticsearch/ingest_pipeline/default.yml:1]: I don't think this should be here.

Comment (MakoWish) [reply packages/winlog/data_stream/winlog/elasticsearch/ingest_pipeline/default.yml:1]: D'oh! I failed to switch branches before I started playing with that one. It has been removed.

File: packages/suricata/data_stream/eve/agent/stream/log.yml.hbs (hunk 2 above)
Comment (MakoWish) [reply packages/suricata/data_stream/eve/agent/stream/log.yml.hbs]: Good point, and done.

File: packages/suricata/data_stream/eve/elasticsearch/ingest_pipeline/default.yml (hunk 3 above)
Comment (MakoWish) [reply packages/suricata/data_stream/eve/elasticsearch/ingest_pipeline/default.yml]: Done

File: packages/suricata/data_stream/eve/_dev/test/pipeline/test-eve-alerts.log-config.yml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+fields:
+  tags:
+    - forwarded
```
Comment (efd6) [inline packages/suricata/data_stream/eve/_dev/test/pipeline/test-eve-alerts.log-config.yml:3]:
```suggestion
    - forwarded
    - preserve_original_event
```

File: packages/suricata/data_stream/eve/_dev/test/pipeline/test-eve-small.log-config.yml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+fields:
+  tags:
+    - forwarded
```
Comment (efd6) [inline packages/suricata/data_stream/eve/_dev/test/pipeline/test-eve-small.log-config.yml:3]:
```suggestion
    - forwarded
    - preserve_original_event
```

**Review by efd6 — COMMENTED** [review]
minor nit and then LGTM

**Comment by MakoWish** [comment]
@efd6 ,

Would you mind kicking off a `/test`?

File: packages/suricata/manifest.yml [hunk 6]
```diff
@@ -1,6 +1,7 @@
 name: suricata
 title: Suricata
-version: "2.12.0"
+version: "2.13.0"
+release: ga
```
Comment (efd6) [inline packages/suricata/manifest.yml:4]:
```suggestion
```

**Comment by efd6** [comment]
You'll need to run `elastic-package test pipeline -g`.

**Comment by MakoWish** [comment]
> You'll need to run `elastic-package test pipeline -g`.

I usually do, but looks like I didn't on this one. Done.

**Review by efd6 — APPROVED** [review]
Thanks

**Pull request merged**